cmd/gateway/azure/gateway-azure.go

```diff
@@ -685,10 +685,22 @@ func (a *azureObjects) GetObjectInfo(ctx context.Context, bucket, object string,
 		return objInfo, azureToObjectError(err, bucket, object)
 	}
 
+	etag := minio.ToS3ETag(blob.Properties.Etag)
```

Comment:
This is not an obvious change @harshavardhana, let's do it in another PR

Reply:
What is not obvious ? This refactor fixes this issue too.